vochain/scrutinizer/scrutinizer.go

```diff
@@ -238,6 +254,16 @@ func (s *Scrutinizer) Commit(height uint32) error {
 		}
 	}
 
+	// Index new transactions
+	for _, tx := range s.newTxPool {
```

Comment:
It would be great if we can run this async, but we should be sure that the code is safe. Also I think we should abstract this code in a function (and just call something like `go updateTxIndex(s.newTxPool)`)

Reply:
I've updated the code to run async, and I've also tested it with the race flag a few times without any races.

Reply:
This update actually was causing many database transaction conflicts. Rather than handling them only for this case, I submitted https://github.com/vocdoni/vocdoni-node/pull/216 which handles transaction conflicts for most update functions. I've rebased this branch on that branch locally and verified that it fixes the problem.

The other issue with running this async is that there's no guarantee that txs will all be indexed after commit is called. For testing, this requires adding a time delay after calling Commit().

Reply:
Why do we have transaction conflicts here? I don't get it. We running the Inserts sync (not in parallel), so there should not be transaction conflicts.

Reply:
The conflict is actually on storing the transaction count. There aren't conflicts within each indexNewTxs call, but there could be multiple calls to this method running at the same time (ie. block 200 has 25 transactions and is indexing them asynchronously, then block 201 has only 1 transaction and indexes it much faster,  catching up with the previous call to indexNewTxs). The conflicts actually stop if I test and only update the txCount if there are txs in the block (rather than updating the txCount from x to x). We should do this anyways to avoid making unnecessary transactions, but the point is a race between transactions is possible here.